mysql-async/src/main/java/com/github/jasync/sql/db/mysql/MySQLConnection.kt

```diff
@@ -246,17 +251,68 @@ class MySQLConnection @JvmOverloads constructor(
         this.serverVersion = parseVersion(message.serverVersion)
         this.serverStatus = message.statusFlags
 
-        this.connectionHandler.write(
-            HandshakeResponseMessage(
-                configuration.username,
-                configuration.charset,
-                message.seed,
-                message.authenticationMethod,
-                database = configuration.database,
-                password = configuration.password,
-                appName = configuration.applicationName
-            )
+        val switchToSsl = when (this.configuration.ssl.mode) {
+            SSLConfiguration.Mode.Disable -> false
+            SSLConfiguration.Mode.Prefer -> message.supportsSSL()
+            SSLConfiguration.Mode.Require,
+            SSLConfiguration.Mode.VerifyCA,
+            SSLConfiguration.Mode.VerifyFull -> {
+                require(message.supportsSSL()) { "SSL is not supported on server" }
+                true
+            }
+        }
+
+        val sslRequest = SSLRequestMessage(setOfNotNull(
+                CapabilityFlag.CLIENT_PLUGIN_AUTH,
```

Comment:
These flags need to be the same for both messages, instead of taking this message as an argument we can also just share the flag set.